### PR TITLE
Set minimum confidence for action

### PR DESCRIPTION
**Pull request by montylion**
This should avoid most, if not all, false positives

### CODE REVIEW

**Comment by ErdbeerbaerLP** [comment]
Does it cuase false negatives tho?

**Comment by montylion** [comment]
> 
> 
> Does it cause false negatives tho?

I did some extensive testing, and it doesn't appear to cause false negatives.

File: src/check-nsfw.js [hunk 1]
```diff
@@ -69,6 +69,7 @@ async function checkNSFW(message, urls) {
 				// aggressive NSFW check
 				// flag the GIF if ANY NSFW frame is found
 				if (frameClassification.className === 'Porn' || frameClassification.className === 'Hentai' || frameClassification.className === 'Sexy') {
+					if (frameClassification.probability < 0.8) return;  // if low confidence then do nothing
```
Comment (RedDuckss) [inline src/check-nsfw.js:72]: Doing a return in the loop will exit the loop and not check further frames, `continue` should be used instead. Also Should probably check the confidence before checking the type?

File: src/check-nsfw.js [hunk 2]
```diff
@@ -87,6 +88,7 @@ async function checkNSFW(message, urls) {
 
 			// check if the prediction is black listed
 			if (classification.className === 'Porn' || classification.className === 'Hentai' || classification.className === 'Sexy') {
+				if (classification.probability < 0.8) return; // if low confidence then do nothing
```
Comment (RedDuckss) [inline src/check-nsfw.js:91]: Also here, should probably check the confidence before the type to save a bit of time. It doesn't matter what the type is if it isn't confident

Comment (montylion) [reply src/check-nsfw.js]: Fixed!

File: src/check-nsfw.js (hunk 1 above)
Comment (montylion) [reply src/check-nsfw.js]: Fixed!

File: src/check-nsfw.js [hunk 3]
```diff
@@ -86,9 +86,9 @@ async function checkNSFW(message, urls) {
 				return (previous.probability > current.probability) ? previous : current;
 			});
 
+			if (classification.probability < 0.8) return;  // if low confidence then do nothing
```
Comment (RedDuckss) [inline src/check-nsfw.js:89]: Just remembered that this check is _also_ in a loop (the bot checks _ALL_ urls it finds in a message) so this should also be `continue`. My bad

Comment (montylion) [reply src/check-nsfw.js]: Oops yeah completely missed that, my bad, I should've checked :/
Anyways, now it's fixed!

**Pull request merged**